java/core/src/java/org/apache/orc/impl/ReaderImpl.java

```diff
@@ -472,8 +473,9 @@ public static OrcTail extractFileTail(ByteBuffer buffer, long fileLength, long m
         .setPostscript(ps)
         .setFooter(footer)
         .setFileLength(fileLength);
+      isCodecError = false;
     } finally {
```

Comment:
try {} finally is pretty broken. You are better off using the try-with-resources statement.

Reply:
existing code